low-level/sensors/impl/centos_7/service_monitor.py

```diff
@@ -0,0 +1,499 @@
+
+    # Dependency list
+    DEPENDENCIES = {"plugins": ["SeviceMsgHandler"]}
+
+    @staticmethod
+    def name():
+        """@return: name of the module."""
+        return ServiceMonitor.SENSOR_NAME
+
+    def __init__(self):
+        """Initialize the relavent datastructures."""
+        super(ServiceMonitor, self).__init__(self.SENSOR_NAME,
+                                                self.PRIORITY)
+
+        Conf.load(self.SSPL_CONFIG, "yaml:///etc/sspl.conf")
+        self.services_to_monitor = \
+            Conf.get(self.SSPL_CONFIG, f"{self.SERVICEMONITOR}>{self.MONITORED_SERVICES}", [])
+
+        self.not_active_services = {}
+        self.failed_services = []
+
+        self.service_status = {}
+
+        self.thread_sleep = \
+            int(Conf.get(self.SSPL_CONFIG, f"{self.SERVICEMONITOR}>{self.THREAD_SLEEP}", 1))
+
+        self.polling_frequency = \
+            int(Conf.get(self.SSPL_CONFIG, f"{self.SERVICEMONITOR}>{self.POLLING_FREQUENCY}", 30))
+
+        self.max_wait_time = \
+            int(Conf.get(self.SSPL_CONFIG, f"{self.SERVICEMONITOR}>{self.MAX_WAIT_TIME}", 60))
+
+    def read_data(self):
+        """Return the dict of service status."""
+        return self.service_status
+
+    def initialize(self, conf_reader, msgQlist, product):
+        """initialize configuration reader and internal msg queues."""
+        # Initialize ScheduledMonitorThread and InternalMsgQ
+        super(ServiceMonitor, self).initialize(conf_reader)
+
+        # Initialize internal message queues for this module
+        super(ServiceMonitor, self).initialize_msgQ(msgQlist)
+
+        # Integrate into the main dbus loop to catch events
+        DBusGMainLoop(set_as_default=True)
+
+        # Initialize SystemBus and get Manager Interface
+        self._bus = SystemBus()
+        systemd = self._bus.get_object("org.freedesktop.systemd1",
+                                       "/org/freedesktop/systemd1")
+        self._manager = Interface(systemd,
+                            dbus_interface='org.freedesktop.systemd1.Manager')
+
+        self.remove_disabled_services()
+
+        return True
+
+    def remove_disabled_services(self):
+        """Remove `disabled` services from the list of services to monitor."""
+        temp = self.services_to_monitor.copy()
+        for service in temp:
+            try:
+                if 'disabled' in str(self._manager.GetUnitFileState(service)):
+                    self.services_to_monitor.remove(service)
+            except DBusException as err:
+                # If a service is enabled then it definitely has 'UnitFileState`,
+                # but for disabled both presence or absence of UnitFileState is
+                # possible. so if `UnitFileState' not present for the service,
+                # it is definitely disabled.
+                logger.error(f"{service} removed from services_to_monitor due "\
+                             f"to error : {err}")
```

Comment:
`Mar 24 00:24:56 localhost sspl-ll[5818]: ERROR lnet.service removed from services_to_monitor due to error : org.freedesktop.DBus.Error.FileNotFound: No such file or directory (service_monitor.py:128)`
`Mar 24 00:24:56 localhost sspl-ll[5818]: ERROR salt-master.service removed from services_to_monitor due to error : org.freedesktop.DBus.Error.FileNotFound: No such file or directory (service_monitor.py:128)`
`Mar 24 00:24:56 localhost sspl-ll[5818]: ERROR salt-minion.service removed from services_to_monitor due to error : org.freedesktop.DBus.Error.FileNotFound: No such file or directory (service_monitor.py:128)`
These Error logs are generated here. I feel these should be warning or info or even debug since we are logging  final list of services we are monitoring on the node in line 133 run() method..

Reply:
you mean duplicated logging happening. here per service and on line 133 in one go? If yes, then yes change this to debug. Also update log statement to something meaningful, someone looking at it should be able to understand without the need to understand the logic or code

Reply:
No duplicate logging is not taking place the services which are disabled and thus DBus Manager is unable to find `UnitStateFile`  then we are logging an error that `this service was removed  from the monitored list` but I think that this should be either info or debug log. Since we are logging about what services we are going to monitor in run() method, so if at some point we are not able to observe a service in the monitored list we can see debug log  instead of always keeping that log as error or warning or info.

Reply:
changed it to logger.debug()